provisioning/miniprov/hare_mp/cdf.py

```diff
@@ -37,11 +38,44 @@ def _create_node_descriptions(self) -> List[NodeDesc]:
             nodes.append(self._create_node(node))
         return nodes
 
+    def _create_pool_descriptions(self) -> List[PoolDesc]:
+        pools: List[PoolDesc] = []
+        conf = self.provider
+        cluster_id = conf.get('cluster>cluster_id')
+        storage_set_count = int(conf.get(f'cluster>{cluster_id}>site>storage_set_count'))
+
+        for x in range(storage_set_count):
+            storage_set_name = conf.get(f'cluster>{cluster_id}>storage_set{x+1}>name')
+            pools.append(PoolDesc(
+                name=Text(storage_set_name),
+                disk_refs=Maybe(DList([
+                    DiskRef(path=Text(device), node=Maybe(Text(node),None))
+                    for node in conf.get(f'cluster>{cluster_id}>storage_set{x+1}>server_nodes')
+                    for device in conf.get(f'cluster>{node}>storage>data_devices')
+                ], 'List Text'), []),
```

Comment:
Note that 'List Text' is a label that will be used as type annotation in Dhall in case if the list becomes empty (Dhall enforces the writer to specify the type explicitly then). Example:
https://github.com/Seagate/cortx-hare/blob/main/provisioning/miniprov/test/test_cdf.py#L47-L53

In your case, the type of the element in the list is defintely not Text but DiskRef. You can write a unit test where this list becomes empty and see how Dhall complains 😉